### PR TITLE
Texture Loading complete

### PR DESCRIPTION
**Pull request by Waterpicker**
Waterpicker Watepricker@rocketmail.com

### CODE REVIEW

File: engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+import org.spout.renderer.gl.Texture.Format;
+import org.spout.renderer.gl.Texture.InternalFormat;
+import org.spout.renderer.gl.Texture.WrapMode;
+import org.spout.renderer.util.CausticUtil;
+import org.spout.renderer.util.Rectangle;
+
+public class SpoutTexture extends Texture {
+	org.spout.renderer.gl.Texture texture;
+	
+	public SpoutTexture(BufferedImage baseImage, Vector2 edge) {
```
Comment (DDoS) [inline engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java]: The edge here is redundant, it's stored in the buffered image

File: engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	public void setComponentType(DataType type) {
+		texture.setComponentType(type);
+	}
+	
+	public void setFormat(Format format) {
+		texture.setFormat(format);
+	}
+	
+	public void setImageData(ByteBuffer buffer, int x, int y) {
+		texture.setImageData(buffer, x, y);
```
Comment (DDoS) [inline engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java]: This should be width and height, not x and y, to avoid confusion.

**Comment by DDoS** [comment]
This still keeps the image data in the memory. You store a buffered image in the texture class, and set the data to the caustic object just after. The caustic object will cleanup the data after creation, but not the texture class. You shouldn't be setting the image data in the caustic object before creation, and after you should set the buffered image to null to let it be GC'ed. Once the data has been uploaded to the GPU after creation, we shouldn't need to have on the memory.

Also, formatting is wrong. You're using Netbeans right? You know there's a key command to format your file automatically... If the formatting Netbeans is using is wrong, you can change it in the preferences.

**Comment by kitskub** [comment]
@DDoS what's wrong with formatting?

File: engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+	}
+	
+	public void setWrapS(WrapMode mode) {
+		texture.setWrapS(mode);
+	}
+	
+	public void setWrapT(WrapMode mode) {
+		texture.setWrapT(mode);
+	}
+	public void unbind() {
```
Comment (DDoS) [inline engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java:124]: This

File: engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+	}
+	
+	public void checkCreated() {
+		texture.checkCreated();
+	}
+	
+	public void create() {
+		texture.create();
+		
+	}
```
Comment (DDoS) [inline engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java:67]: This

Comment (lukespragg) [inline engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java:67]: New line not needed at line 69.

File: engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java (hunk 3 above)
Comment (lukespragg) [inline engine/src/main/java/org/spout/engine/filesystem/resource/SpoutTexture.java:124]: New line needed after line 125, before 126.

**Pull request merged**